data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -276,6 +276,20 @@ public String toJsonString() {
         return jsonNode.toString();
     }
 
+    @Override
+    public String toJsonStringWithTags(final String tagsKey) {
```

Comment:
I'm not sure this is a very extensible approach. Perhaps some form of JSON string builder would be more appropriate.

As a caller, perhaps I could call:

```
event.jsonStringBuilder()
  .includeTags("tags")
  .toJsonString()
```

Over time we may find other features that we want to include in the event.